PHPCompatibility/Tests/Syntax/NewConstantDereferencingUnitTest.inc

```diff
@@ -0,0 +1,36 @@
+<?php
+
+// OK: non-constants using array or object dereferencing
+echo $a[0];
+echo $a{0};
+echo $a->length();
+
+// OK: references to non-constants
+class Foo { }
+echo $foo::$bar[0];
+class Bar extends Foo { }
+class Bar implements Foo { }
+interface Foo { }
+namespace Foo { }
+echo $foo->bar[0];
+trait Foo { }
+use Foo as Bar;
+$foo = new Foo;
+function foo() { }
+if ($foo instanceof Foo) { }
+use A, B { B::foo insteadof A };
+goto foo;
+private Foo $foo;
+protected Foo $foo;
+public Foo $foo;
+
```

Comment:
Quite a few of the above tests are not testing anything or testing the same thing.

Reply:
I think I've eliminated the redundant tests. If there are any cases you still feel are unnecessary, please let me know.